tools/commit_slider/case_1.md

```diff
@@ -0,0 +1,316 @@
+# Case #1. Cross-checking of performance degradation.
+#### Themes: Mode and Template customization, API design
+  
+
+For 2 given versions, defined by commit hashes and 2 models we need to resolve which of two changes (ov or model) caused degradation. I.e. we need to differentiate 2 possible cases:
+model-caused
+|'Bad' model | model_1.xml| model_2.xml|
+| --------     | -------    | -------|
+| ov_version_1 | 100 % (ref)|50%     |
+| ov_version_2 | 98 %       |51%     |
+and ov - caused
+|'Bad' ov-version| model_1.xml| model_2.xml|
+| --------     | -------    | -------|
+| ov_version_1 | 100 % (ref)|98%     |
+| ov_version_2 | 48% %       |51%     |
+
+## Version 1. Reusing of existing classes
+
+### Traversal
+We need full bypass of commit set (pair of commits in the given case), ignoring the result of a specific commit.
+
+It corresponds to the existing `brute_force` traversal.
+
+### Mode
+
+On every commit we print several commands.
+
+For the first version, we also reuse the existing `Nop` (No Operation) mode with simple list of bash commands.
+
+### Configuration
+The simplest configuration with default settings (openvino/, build/ and bit/intel64/Release/ paths correspondingly).
+```
+We use explicitList instead of commit interval.
+{
+    "appCmd" : [
+        "./benchmark_app -m model_1_path.xml",
+        "./benchmark_app -m model_2_path.xml"
+        ],
+    "runConfig" : {
+        "commitList" : {
+            "explicitList" : [ "<start_commit>", "<end_commit>" ]
+        },
+    "mode" : "nop",
+    "traversal" : "bruteForce"
+}
+```
+Or more specified and extended:
+```
+{
+    "appCmd" : [
+        "echo first model",
+        "./benchmark_app -m model_1_path.xml",
+        "echo second model",
+        "./benchmark_app -m model_2_path.xml"
+        ],
+    "appPath": "<path_to_benchmark_app>",
+    "gitPath" : "<path_to_git_repository>",
+    "buildPath" : "<path_to_build_dir>",
+        "runConfig" : {
+            "commitList" : {
+                "explicitList" : [ "<start_commit>", "<end_commit>" ]
+            },
+    "mode" : "nop",
+    "traversal" : "bruteForce"
+}
+```
+### Example of launch and output
+```
+python3 commit_slider.py -cfg cfg.json
+
+current commit: <start_commit_hash>
+first model
+<...> Throughput: 500 FPS
+second model
+<...> Throughput: 500 FPS
+
+current commit: <end_commit_hash>
+first model
+<...> Throughput: 1000 FPS
+second model
+<...> Throughput: 1000 FPS
+
+```
+It faces given problem, but looks clumsy and verbose. We also need to interpret results ourselves.
+
+## Second Version: Specific Mode
+
+If there is no Mode which resolves commit with our task or we need more readable output and opportunity to handle every commit directly, we can implement custom Mode. We can also customize API.
+### Mode
+```
+# Mode suppose to be as much generic as possible and correspond all similar problems so not to be not connected with the separate user case, but as the example we provide simplified version.
```

Comment:
Please write a comment on a few lines to avoid horizontal scrolling. The same for others long comments